### PR TITLE
Severin Wiggenhorn - Carets 

### PR DESCRIPTION
**Pull request by s-wigg**
# Media Ranker
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
---      | ---
Describe a custom model method you wrote. | I wrote a top_ten work model method that takes an input of a category name and then finds up to ten works that have the most votes in that category|
Describe how you approached testing that model method. What edge cases did you come up with? | Because I used that model to generate the view on the home/top media page I needed to make sure it would supply works even if none of them had any votes. I generated a bunch of works with no votes, applied the top_ten method to them, saved it to the variable and checked to make sure the variable.size was 10. I also checked that if there are fewer than 10 of a certain category, it will get all of the works there are. |
Describe an edge case test you wrote for a controller |In the user controller, I included a test, where if the url includes an invalid user id (i.e. -2) that it will do a 404 :not_found error|
What are `session` and `flash`? What is the difference between them? |Session allows information to persist in a hash-like object until the browser is closer (or the session information is otherwise set to nil). Session can be used, for example, to track user log in and log out. Flash allows information to be transmitted, for example, from the controller to the view and can be be used to print messages such as that something was successfully destroyed or edited. |
Describe a controller filter you wrote. | I used a controller filter in the works controller for show, edit, update, and destroy, that finds the work based on the work id|
What was one thing that you gained more clarity on through this assignment? | Adding custom model methods and using nested routes|
What is the Heroku URL of your deployed application | https://best-ever-media-ranker.herokuapp.com/ |
Do you have any recommendations on how we could improve this project for the next cohort? | I think this project would have been better as the pair project and then ride share could have been an individual project. This was also a lot to get done in a 4 day week. I think starting to introduce testing last week on ride share would have made one fewer new thing to try to figure out. Also, the lecture on how to use foundation should have included some actual code examples, instead of purely conceptual discussions about mobile first. |
,

### CODE REVIEW

**Comment by tildeee** [comment]
# Media Ranker
## What We're Looking For

Feature | Feedback
---     | ---
**Core Requirements** |
Git hygiene | x
Comprehension questions	| x
**General** |
Rails fundamentals (RESTful routing, use of named paths) | x
Semantic HTML | x
Errors are reported to the user | x
Business logic lives in the models | x
Models are thoroughly tested, including relations, validations and any custom logic | x
Controllers are thoroughly tested | x
**Wave 1 - Media** |
Splash page shows the three media categories | x
Basic CRUD operations on media are present and functional | x
**Wave 2 - Users and Votes** |
Users can log in and log out | x
The ID of the current user is stored in the session | x
Individual user pages and the user list are present | x
A user cannot vote for the same media more than once | x
All media lists are ordered by vote count | x
Splash page contains a media spotlight | x
Media pages contain lists of voting users | x
**Wave 3 - Styling** |
Foundation is used appropriately | x
Look and feel is similar to the original | x
**Overall** |

**Comment by tildeee** [comment]
Good job! The site works really well and has all the functionality. Good job on the tests, too

Also, best heroku app name ever

Just a few comments:
- The form for works doesn't prevent input in publication year as a non-integer, so if you save a non-integer into the form, it becomes 0
- If you just put in white space into the login name for the login form, it doesn't redirect

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        <th>Username</th>
+        <th width="100">Votes</th>
+        <th width="200">Joined</th>
+      </tr>
+    </thead>
+    <tbody>
+      <% @users.each do |user| %>
+          <tr>
+          <td><%= link_to(user.name, user_path(user.id), alt: "link to user show page")%></td>
+          <td> # </td>
```
Comment (tildeee) [inline app/views/users/index.html.erb]: Not sure why this isn't filled out?
